pygmt/src/grdcut.py

```diff
@@ -91,10 +98,21 @@ def grdcut(grid, outgrid: str | None = None, **kwargs):
     >>> # 12° E to 15° E and a latitude range of 21° N to 24° N
     >>> new_grid = pygmt.grdcut(grid=grid, region=[12, 15, 21, 24])
     """
+    inkind = data_kind(grid)
+    outkind = "grid"
+    if inkind == "image":
+        outkind = "image"
+    if inkind == "file":
+        realpath = which(str(grid))
+        if isinstance(realpath, list):
+            realpath = realpath[0]
+        if realpath.endswith(".tif"):
+            outkind = "image"
+
     with Session() as lib:
         with (
             lib.virtualfile_in(check_kind="raster", data=grid) as vingrd,
-            lib.virtualfile_out(kind="grid", fname=outgrid) as voutgrd,
+            lib.virtualfile_out(kind=outkind, fname=outgrid) as voutgrd,
         ):
             kwargs["G"] = voutgrd
             lib.call_module(
```

Comment:
Ok, so if we are able to have `outkind="image"` or `outkind="grid"`, should we rename the `virtualfile_to_grid` method to something like `virtualfile_to_grid_or_image`? Since that method should be able to handle both 1-band grids and 3-band images.

Reply:
You're talking about the `virtualfile_to_grid` method implemented in #2398, right? The method looks like this:
```python
def virtualfile_to_grid(self, vfname, outgrid):
    if outgrid is not None:
        return None
    return self.read_virtualfile(vfname, kind="grid").contents.to_dataarray()
```
Yes, the same code can be used for grid, image and even cubes. We just need to pass the correct `kind` parameter and make sure that `GMT_GRID`/`GMT_IMAGE`/`GMT_CUBE` all have the `.to_dataarray()` method.

So we can rewrite the method to:
```
def virtualfile_to_raster(self, vfname, outgrid, kind="grid"):
    if outgrid is not None:
        return None
    return self.read_virtualfile(vfname, kind=kind).contents.to_dataarray()
```
Actually, as mentioned in https://github.com/GenericMappingTools/pygmt/pull/3115/files#r1536468908, it's possible to determine the virtual file family by calling `GMT_Inquire_Family`.

So, what we should do is:

1. Wrap the `GMT_Inquire_Family`
2. Refactor the `read_virtualfile` method. Currently, `kind=None` means no casting into a GMT_* data structure. We can change the behavior to: if `kind=None`, then we call `GMT_Inquire_Family` to determine the virtual file family and then cast it to the corresponding GMT_* data structure.
3. Then the method can be simplified to:
```
def virtualfile_to_raster(self, vfname, outgrid):
    if outgrid is not None:
        return None
    return self.read_virtualfile(vfname).contents.to_dataarray()
```